pkg/kubelet/cm/node_container_manager_linux.go

```diff
@@ -53,6 +53,29 @@ func (cm *containerManagerImpl) createNodeAllocatableCgroups() error {
 	return nil
 }
 
+// UpdateNodeAllocatableCgroups updates cgroups for node allocatable resources.
+func (cm *containerManagerImpl) UpdateNodeAllocatableCgroups(rl v1.ResourceList) error {
```

Comment:
I think I prefer to split the logic of this function into two, to make the side effects more visible.

What if you change the signature to (I have no idea about the naming scheme):
```go
func (cm *containerManagerImpl) UpdateNodeAllocatableCgroups(rl v1.ResourceList) bool {
    [...]
    return changed
}
```
instead, so that the "user" can run `cm.enforceNodeAllocatableCgroups`?

Another alternative can be to split the "check if this have changed" into one func nad "update the internalCapacity" into another one.

Both these approaches will make it way easier to test too.